packages/stream-api/src/streamApi.ts

```diff
@@ -38,78 +43,125 @@ export class StreamApi {
 
   public startStream(
     data: vertexvis.protobuf.stream.IStartStreamPayload
-  ): Promise<void> {
+  ): Promise<void>;
+
+  public startStream(
+    requestId: UUID.UUID,
+    data: vertexvis.protobuf.stream.IStartStreamPayload
+  ): Promise<vertexvis.protobuf.stream.IStartStreamResult>;
+
+  public startStream(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
     return this.sendRequest({
+      ...request.requestId,
       startStream: {
-        ...data,
+        ...request.data,
       },
     });
   }
 
-  public beginInteraction(): Promise<void> {
+  public beginInteraction(): Promise<void>;
+
+  public beginInteraction(
+    requestId: UUID.UUID
+  ): Promise<vertexvis.protobuf.stream.IBeginInteractionResult>;
+
+  public beginInteraction(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
     return this.sendRequest({
+      ...request.requestId,
       beginInteraction: {},
     });
   }
 
   public replaceCamera({
     camera,
-  }: vertexvis.protobuf.stream.IUpdateCameraPayload): Promise<void> {
-    return this.sendRequest({
+  }: vertexvis.protobuf.stream.IUpdateCameraPayload): Promise<void>;
+
+  public replaceCamera(
+    requestId: UUID.UUID,
+    { camera }: vertexvis.protobuf.stream.IUpdateCameraPayload
+  ): Promise<vertexvis.protobuf.stream.IUpdateCameraResult>;
+
+  public replaceCamera(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
+    const requestPayload = {
+      ...request.requestId,
       updateCamera: {
-        camera,
+        camera: request.data as vertexvis.protobuf.stream.ICamera,
       },
-    });
+    };
+    return this.sendRequest(requestPayload);
   }
 
+  public hitItems({
+    point,
+  }: vertexvis.protobuf.stream.IHitItemsPayload): Promise<void>;
+
   public hitItems(
     requestId: string,
     { point }: vertexvis.protobuf.stream.IHitItemsPayload
-  ): Promise<vertexvis.protobuf.stream.IStreamResponse> {
-    return this.sendRequestWithId({
-      requestId: {
-        value: requestId,
-      },
+  ): Promise<vertexvis.protobuf.stream.IStreamResponse>;
+
+  public hitItems(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
+    return this.sendRequest({
+      ...request.requestId,
       hitItems: {
-        point,
+        point: request.data as vertexvis.protobuf.stream.IPoint,
       },
     });
   }
 
-  public endInteraction(): Promise<void> {
+  public endInteraction(): Promise<void>;
+
+  public endInteraction(
+    requestId: UUID.UUID
+  ): Promise<vertexvis.protobuf.stream.IBeginInteractionResult>;
+
+  public endInteraction(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
     return this.sendRequest({
+      ...request.requestId,
       endInteraction: {},
     });
   }
 
+  private parseArgs(args: unknown[]): ParseArgsRequest {
```

Comment:
Consider making this a generic method, this way you don't have to do explicit casting.

```ts
private parseArgs<T>(args: unknown[]): ParseArgsRequest[T] {
  ...
}

public hitItems(...args: unknown[]): unknown {
  const req = this.parseArgs[vertexvis.protobuf.stream.IHitItemsPayload](args);
  req.data; // vertexvis.protobuf.stream.IPoint
}
```

Reply:
is the `[T]` mapping to `data` somehow? it's giving me an error when I use that syntax, not finding examples or docs on using in that way.

Reply:
worked on this here 2a8ab2ecd65df1f22f09543f94dc67b02f34f798